tests/integration/clientv3/concurrency/session_test.go

```diff
@@ -111,3 +112,76 @@ func TestSessionCtx(t *testing.T) {
 	}
 	assert.Equal(t, childCtx.Err(), context.Canceled)
 }
+
+// TestSessionCreationTimeout checks that the option WithSessionCreationTimeout
+// sets a timeout for the creation of new sessions
+func TestSessionCreationTimeout(t *testing.T) {
```

Comment:
I think this case covered the block case. 

If we need to confirm that the underlying grpc is retrying to get connection, as far as I know, we can use logger observer to expect the log content like this.

<details>

<summary> example </summary>

```diff
diff --git a/tests/framework/integration/testing.go b/tests/framework/integration/testing.go
index b0580dcd6..4a839668d 100644
--- a/tests/framework/integration/testing.go
+++ b/tests/framework/integration/testing.go
@@ -20,6 +20,7 @@ import (

        grpc_logsettable "github.com/grpc-ecosystem/go-grpc-middleware/logging/settable"
        "github.com/stretchr/testify/require"
+       "go.uber.org/zap"
        "go.uber.org/zap/zapcore"
        "go.uber.org/zap/zapgrpc"
        "go.uber.org/zap/zaptest"
@@ -30,6 +31,7 @@ import (
        "go.etcd.io/etcd/client/pkg/v3/verify"
        clientv3 "go.etcd.io/etcd/client/v3"
        "go.etcd.io/etcd/server/v3/embed"
+       "go.etcd.io/etcd/tests/v3/framework/testutils"
 )

 var grpc_logger grpc_logsettable.SettableLoggerV2
@@ -131,6 +133,26 @@ func BeforeTest(t testutil.TB, opts ...TestOption) {
        os.Chdir(t.TempDir())
 }

+func ClientGRPCLoggerObserver(t testutil.TB) *testutils.LogObserver {
+       level := zapcore.InfoLevel
+
+       obCore, logOb := testutils.NewLogObserver(level)
+
+       options := zaptest.WrapOptions(
+               zap.WrapCore(func(oldCore zapcore.Core) zapcore.Core {
+                       return zapcore.NewTee(oldCore, obCore)
+               }),
+       )
+
+       grpc_logger.Set(
+               zapgrpc.NewLogger(
+                       zaptest.NewLogger(t, zaptest.Level(level), options).
+                               Named("grpc-observer"),
+               ),
+       )
+       return logOb
+}
+
 func assertInTestContext(t testutil.TB) {
        if !insideTestContext {
                t.Errorf("the function can be called only in the test context. Was integration.BeforeTest() called ?")
diff --git a/tests/integration/clientv3/concurrency/session_test.go b/tests/integration/clientv3/concurrency/session_test.go
index 247c777df..b9410a93d 100644
--- a/tests/integration/clientv3/concurrency/session_test.go
+++ b/tests/integration/clientv3/concurrency/session_test.go
@@ -119,31 +119,30 @@ func TestSessionCreationTimeout(t *testing.T) {
        integration2.BeforeTest(t)

        // create new cluster
-       clus := integration2.NewCluster(t, &integration2.ClusterConfig{
-               Size: 3,
-       })
-
-       eps := []string{clus.Members[0].GRPCURL(), clus.Members[1].GRPCURL(), clus.Members[2].GRPCURL()}
-       lead := clus.WaitLeader(t)
+       clus := integration2.NewCluster(t, &integration2.ClusterConfig{Size: 1})

        // create new client
-       cli, err := integration2.NewClient(t, clientv3.Config{Endpoints: []string{eps[lead]}})
+       cli, err := integration2.NewClient(t, clientv3.Config{Endpoints: []string{clus.Members[0].GRPCURL()}})
        if err != nil {
+               clus.Terminate(t)
                t.Fatal(err)
        }
        defer cli.Close()

-       // wait for eps[lead] to be pinned
+       // ensure the connection is established.
        clientv3test.MustWaitPinReady(t, cli)

-       // add all eps to list, so that when the original pined one fails
-       // the client can switch to other available eps
-       cli.SetEndpoints(eps...)
-
+       // terminating the cluster
        clus.Terminate(t)

-       _, err = concurrency.NewSession(cli, concurrency.WithSessionCreationTimeout(50))
+       // override the grpc logger
+       logOb := integration2.ClientGRPCLoggerObserver(t)
+
+       _, err = concurrency.NewSession(cli, concurrency.WithSessionCreationTimeout(3000))
        assert.Equal(t, err, context.DeadlineExceeded)
+
+       _, err = logOb.Expect(context.Background(), "Subchannel Connectivity change to TRANSIENT_FAILURE", 3)
+       assert.Nil(t, err)
 }

 // TestTimeoutDoesntAffectSubsequentConnections checks that the option WithSessionCreationTimeout
```

</details>

Reply:
gotcha, applying diff and removing the subsequent test case :+1: